ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceDownloadManager.swift

```diff
@@ -475,4 +487,104 @@ public class ResourceDownloadManager {
     let (lm, _) = downloader.installLexicalModelPackage(downloadedPackageFile: packageURL)
     return lm
   }
+
+  // MARK - Completion handlers.
+
+  internal func resourceDownloadStartClosure<Resource: LanguageResource>(for resources: [Resource]) -> (() -> Void) {
+    return { self.resourceDownloadStarted(for: resources) }
+  }
+
+  // Only for use with individual downloads.  Updates should have different completion handling.
+  internal func resourceDownloadCompletionClosure<Resource: LanguageResource>(for resources: [Resource], handler: CompletionHandler<Resource>?) -> CompletionHandler<Resource> {
```

Comment:
By constructing completion closures within `ResourceDownloadManager` for use by `DownloadBatch`es, note that we're able to move a _lot_ of the resource-tracking data management out of the batch itself.  (These closures don't have access to the batches that will own them.)

In fact, the only real reason to keep all of the "resource" tagging on `DownloadTask` and `DownloadBatch` (after resource updates are also refactored) will be for two related methods in `ResourceDownloadManager`:
- `stateForKeyboard(withID:)`
- `stateForLexicalModel(withID:,languageID:)`
    - These two methods (currently) check against the queue to know if a resource is in the midst of downloading.

If not for that, `DownloadTask` could be non-generic again and further simplified, and nearly all of the resource-tracking stuff in `DownloadTask` and `DownloadBatch` could be removed entirely.